clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1774,6 +1774,18 @@ llvm::Constant *ConstantEmitter::emitForMemory(CodeGenModule &CGM,
     return Res;
   }
 
+  if (const auto *BIT = destType->getAs<BitIntType>()) {
+    if (BIT->getNumBits() > 128) {
+      // Long _BitInt has array of bytes as in-memory type.
+      ConstantAggregateBuilder Builder(CGM);
+      llvm::Type *DesiredTy = CGM.getTypes().ConvertTypeForMem(destType);
+      auto *CI = cast<llvm::ConstantInt>(C);
```

Comment:
I'm not sure this cast is guaranteed to succeed?  At least in some cases, we emit constant expressions involving a ptrtoint.  Maybe at the widths in question, that can't happen, but this deserves a comment explaining what's going on.

Reply:
I've added a comment. I'm not able to get a ptrtoint in a constant expression involving a big _BitInt.

Reply:
How about a "small" `_BitInt` ? The comment starts
> // LLVM type doesn't match AST type only for big enough _BitInts,
and for AArch32 and AArch64 we are going to have a non-matching LLVM types even for "small" `_BitInt`s - for AArch32 because the ABI wants the padding bing in-memory representation to contain zero or the sign-bit, and for both we'd like to emit loads/stores in bigger chunks, e.g. `i17` is a single `i32` load store, as opposed to two separate accesses to `i16` and `i8`.

Reply:
The test case here is just going to be something like `_SomeSplitBitIntType x = (unsigned long) &someVariable;`.  What code do we actually produce for this?  Sometimes we'll be able to fall back on dynamic initialization, but that's not always an option.

Ideally, it's just invalid to do something like that.  It certainly needs to be diagnosed if the integer type is narrower than the pointer, and wider is also problematic, although less so and in a different way.

Reply:
Well, it seems it doesn't depend on the size of _BitInt. Using something like `_SomeSplitBitIntType x = (unsigned long) &someVariable;` either fails if I for example apply `constexpr` or falls back on dynamic initialization. So I changed the comment to make it more generic.